RAG_supporters/dataset/steering_embedding_generator.py

```diff
@@ -0,0 +1,286 @@
+)
+from RAG_supporters.clustering.clustering_data import ClusteringData
+from RAG_supporters.dataset.steering.steering_config import SteeringConfig, SteeringMode
+
+
+class SteeringEmbeddingGenerator:
+    """
+    Generates steering embeddings based on configured modes.
+    
+    Supports:
+    - SUGGESTION: Uses suggestion embeddings
+    - CLUSTER_DESCRIPTOR: Uses cluster descriptor embeddings
+    - LLM_GENERATED: Uses LLM-generated steering text embeddings
+    - ZERO: Returns zero vector (no steering)
+    - MIXED: Weighted combination of modes
+    
+    Integrates augmentations from embedding.py for noise injection.
+    """
+    
+    def __init__(
+        self,
+        config: SteeringConfig,
+        clustering_data: ClusteringData,
+        embedding_model: Any,  # Model with encode() method
+        suggestion_embeddings: Optional[Dict[str, np.ndarray]] = None,
+        llm_steering_texts: Optional[Dict[str, str]] = None,
+        augment_noise_prob: float = 0.0,
+        augment_zero_prob: float = 0.0,
+        augment_noise_level: float = 0.01
+    ):
+        """
+        Initialize steering embedding generator.
+        
+        Args:
+            config: Steering configuration with modes
+            clustering_data: Cluster data with centroids and descriptors
+            embedding_model: Model for encoding text to embeddings
+            suggestion_embeddings: Pre-computed suggestion embeddings {suggestion: embedding}
+            llm_steering_texts: LLM-generated steering texts {key: text}
+            augment_noise_prob: Probability of applying noise augmentation
+            augment_zero_prob: Probability of zeroing steering embedding
+            augment_noise_level: Std deviation for noise augmentation
+        """
+        self.config = config
+        self.clustering_data = clustering_data
+        self.embedding_model = embedding_model
+        self.suggestion_embeddings = suggestion_embeddings or {}
+        self.llm_steering_texts = llm_steering_texts or {}
+        self.augment_noise_prob = augment_noise_prob
+        self.augment_zero_prob = augment_zero_prob
+        self.augment_noise_level = augment_noise_level
+        self.rng = random.Random(config.random_seed)
+    
+    def generate(
+        self,
+        sample_id: int,
+        suggestions: Optional[List[str]] = None,
+        cluster_id: Optional[int] = None
+    ) -> Tuple[np.ndarray, SteeringMode]:
+        """
+        Generate steering embedding for a sample.
+        
+        Args:
+            sample_id: Sample identifier
+            suggestions: List of suggestion strings
+            cluster_id: Primary cluster ID for descriptor mode
+            
+        Returns:
+            Tuple of (steering_embedding, mode_used)
+        """
+        # Select mode based on configured probabilities
+        mode = self._select_mode()
+        
+        # Generate embedding based on mode
+        if mode == SteeringMode.ZERO:
```

Comment:
Instead of if-elif chain, use dictionary dispatch: self._generators = {
    SteeringMode.ZERO: self._generate_zero_embedding,
    SteeringMode.SUGGESTION: self._generate_suggestion_embedding,
    # ...
}